src/Microsoft.ApplicationInsights/DataContracts/ExceptionInfo.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Extensibility.Implementation;
+    using Extensibility.Implementation.External;
+
+    /// <summary>
+    /// Wrapper class for <see cref="ExceptionData"/> that lets user provide exception data without having the actual Exception object.
+    /// </summary>
+    internal sealed class ExceptionInfo
+    {
+        private readonly ExceptionData data;
+        
+        /// <summary>
+        /// Constructs the instance of <see cref="ExceptionInfo"/>
+        /// </summary>
+        public ExceptionInfo(IEnumerable<ExceptionDetailsInfo> exceptionDetailsInfoList, SeverityLevel? severityLevel, string problemId,
+            IDictionary<string, string> properties, IDictionary<string, double> measurements)
+        {
+            this.data = new ExceptionData
+            {
+                exceptions = exceptionDetailsInfoList.Select(edi => edi.ExceptionDetails).ToList(),
+                severityLevel = severityLevel.TranslateSeverityLevel(),
+                problemId = problemId,
+                properties = new ConcurrentDictionary<string, string>(properties),
+                measurements = new ConcurrentDictionary<string, double>(measurements)
+            };
+        }
+
+        internal ExceptionInfo(ExceptionData data)
+        {
+            this.data = data;
+        }
+
+        /// <summary>
+        /// Gets a list of <see cref="ExceptionDetailsInfo"/> to modify as needed.
+        /// </summary>
+        public IReadOnlyList<ExceptionDetailsInfo> ExceptionDetailsInfoList => this.data.exceptions.Select(ed => new ExceptionDetailsInfo(ed)).ToList().AsReadOnly();
```

Comment:
why is it readonly? It is OK to allow people to remove unnecessary inner exceptions

Reply:
See the comment above. Because we're wrapping objects into the publicly facing type, we'd have to jump through hoops in order to make this happen. Kunal's scenario doesn't require this, so I suggest until someone actually needs it we don't spend time here.

Reply:
I think this is an important scenario. It may save a lot of money to customer. Please create a separate issue to track it.